src/event_handler.rs

```diff
@@ -504,63 +511,148 @@ pub async fn handle_event_logs(
+            let http_rpc_client = app_state.http_rpc_client.lock().unwrap().clone().unwrap();
+            let Ok(txn_receipt) = http_rpc_client
+                .get_transaction_receipt(pending_txn_data.txn_hash)
+                .await
+            else {
+                continue;
+            };
+
+            // Transaction is confirmed/mined and need not be resent
+            if txn_receipt.is_some() {
+                retry_txn = false;
+                break;
+            }
+
+            // Current txn is dropped/lost and need to be resent with higher gas price
+            pending_txn_data.gas_limit = pending_txn_data.gas_limit + GAS_LIMIT_BUFFER;
+            pending_txn_data.gas_price = U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
+                * pending_txn_data.gas_price
+                / 100;
+
+            // Send the replacement txn if the current one is dropped or lost
+            while Instant::now() < pending_txn_data.retry_deadline {
+                let replacement_txn = pending_txn_data.txn_data.clone();
+                let replacement_txn = replacement_txn
+                    .nonce(pending_txn_data.nonce)
+                    .gas_price(pending_txn_data.gas_price)
+                    .gas(pending_txn_data.gas_limit);
+
+                let pending_txn = replacement_txn.send().await;
+                let Ok(pending_txn) = pending_txn else {
+                    let error_string = format!("{:?}", pending_txn.unwrap_err());
+
+                    // Retry after a delay if connection failed
+                    if error_string.contains("code: -32000")
+                        && (error_string.contains("connection") || error_string.contains("network"))
+                    {
+                        sleep(Duration::from_millis(200)).await;
+                        continue;
+                    }
+
+                    // Retry with a higher gas price if the replacement txn is underpriced
+                    if error_string.contains("code: -32000")
+                        && (error_string.contains("replacement transaction underpriced")
+                            || error_string.contains("transaction underpriced"))
+                    {
+                        pending_txn_data.gas_price =
+                            U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
+                                * pending_txn_data.gas_price
+                                / 100;
+
+                        sleep(Duration::from_millis(10)).await;
+                        continue;
+                    }
+
+                    // Break in case of other errors like "nonce too low" or "execution reverted" or "insufficient funds" etc.
+                    retry_txn = false;
+                    break;
+                };
+
+                // Monitor the newly sent pending txn
+                pending_txn_data.txn_hash = pending_txn.tx_hash();
+                break;
```

Comment:
don't you want to put this pending_txn_data back on top of deque, and do the same process till deadline is elapsed. If deadline is elapsed for top pending_txn ,then only send the dummy txn. Please check the flow and every other case.

Reply:
The way I'm handling the pending txns queue is by popping the front txn (least nonce txn), focussing on sending the txn for that nonce (either the replacement until the deadline is met and dummy afterwards) and after the current pending nonce is successfully mined move on to the next txn and so on.

Reply:
Okay, it's just exiting from the inner loop only. Now got it.